src/test/scala/latis/input/HapiBinaryAdapterSpec.scala

```diff
@@ -0,0 +1,47 @@
+package latis.input
+
+import java.net.URI
+
+import latis.data.{Data, Sample, SeqFunction}
+import latis.metadata.Metadata
+import latis.model._
+import latis.output.TextWriter
+import latis.util.FileUtils
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers._
+
+class HapiBinaryAdapterSpec extends FlatSpec {
+
+  val reader = new AdaptedDatasetReader {
+    //def uri: URI = new URI("https://cdaweb.gsfc.nasa.gov/hapi/data?id=AC_H0_MFI&time.min=2019-01-01&time.max=2019-01-02&parameters=Magnitude,dBrms&format=binary")
+    //def uri: URI = FileUtils.resolvePath("data/hapi_binary_data").get.toUri
+    def uri: URI = new URI(s"file:${System.getProperty("user.home")}/git/latis3-hapi/src/test/resources/data/hapi_binary_data")
+    def model: DataType = Function(
+      Scalar(Metadata("id" -> "Time", "type" -> "string", "length" -> "24")),
+      Tuple(
+        Scalar(Metadata("id" -> "Magnitude", "type" -> "float")),
+        Scalar(Metadata("id" -> "dBrms", "type" -> "float"))
+      )
+    )
+    def adapter = new HapiBinaryAdapter(model)
+  }
+
+  val ds = reader.getDataset
+  //TextWriter().write(ds)
+
+  "The first sample in the HAPI Binary dataset" should "contain the correct values" in {
+    ds.unsafeForce.data match {
+      case sf: SeqFunction => sf.samples.head match {
+        case Sample(d, r) => (d, r) match {
+          case (List(time), List(mag, dbrms)) =>
```

Comment:
The test matching idioms are evolving. We can now do `StreamUtils.unsafeHead(ds.samples)` to get the first sample to match on like: `case Sample(DomainData(Text(time)), RangeData(Real(mag), Real(db)))`.
I'm open for suggestions. I'm guessing the we could create some sort of test DSL and/or take advantage of ScalaTest Matchers one of these days.

Reply:
Will you please make this a requested change so I don't forget to address it?

Reply:
After pulling in the latest changes to `latis3` master, I can no longer build the project. I get this error in `latis.input.HttpStreamSource`: `object http4s is not a member of package org`. I confirmed that `http4s` is in the `build.sbt` and I used the command line to pull in the `http4s` dependency (and saw new dependencies for `http4s` get brought in. But `HttpStreamSource` still won't compile. Did either of you encounter this problem?

Reply:
Does it build with `sbt compile`? Maybe intelliJ needs to be told to autoimport when the sbt build changes?

Reply:
Huh, no it does not. When I run `./sbt compile` I get errors related to: `Dap2Service.scala:14:8: object Dataset is not a member of package latis.model` which is even more curious.

Reply:
How about `./sbt core/compile`?
I need to fix that dap subproject.

Reply:
Okay that worked. I guess it must be an IntelliJ thing.